### PR TITLE
from_Solis

### LINKED ISSUE #656: `from_andor` method
generate `Data` object from ascii exports of CCD

### PR DESCRIPTION
**Pull request by p770193**
Method for importing Andor spectrometer CCD data (exported ascii format).  

resolves #656 

- [x] write from function
- [x] add example dataset(s), describe in docs
- [x] write test(s)

### CODE REVIEW

File: WrightTools/data/_andor.py [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+        Default is None.
+    parent : WrightTools.Collection (optional)
+        Collection to place new data object within. Default is None.
+    verbose : boolean (optional)
+        Toggle talkback. Default is True.
+
+    Returns
+    -------
+    data
+        New data object(s).
```
Comment (untzag) [inline WrightTools/data/_andor.py]: this function should always return one data object---clarify that here

File: WrightTools/data/_andor.py [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+        axis0 = []
+        arr = []
+        attrs = {}
+        while True:
+            line = f.readline().strip()[:-1]
+            if len(line) == 0:
+                break
+            else:
+                line = line.split(',')
+                line = [eval(x) for x in line]
```
Comment (untzag) [inline WrightTools/data/_andor.py]: `eval` considered harmful...

do we need `eval`? I've just got to imagine that there is a better way to parse this info

at the very least, can we use [`ast.literal_eval`](https://docs.python.org/2/library/ast.html#ast.literal_eval) instead?

**Comment by p770193** [comment]
@wright-group/wrighttools-maintainers ready for re-review.

File: WrightTools/data/__init__.py [hunk 3]
```diff
@@ -26,6 +27,7 @@
     "Channel",
     "Variable",
     # From methods in alphabetic order
+    "from_andor",
```
[the remark this answers is not in the document]
Comment (untzag) [reply WrightTools/data/__init__.py]: this actually breaks import, so I cannot test how the docs work right now---next round :smile:

[the remark this answers is not in the document]
Comment (p770193) [reply WrightTools/data/__init__.py]: sorry about that...should be good now

File: WrightTools/data/_solis.py [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+
+def from_Solis(filepath, name=None, parent=None, verbose=True):
+    """Create a data object from Andor Solis software (ascii exports).
+
+    Parameters
+    ----------
+    filepath : string, list of strings, or array of strings
+        Path to .txt file.
+    name : string (optional)
+        Name to give to the created data object. If None, filename is used.
```
Comment (ksunden) [inline WrightTools/data/_solis.py]: Does file metadata provide a name? if so, that is preferred to the file name itself

File: WrightTools/data/_solis.py [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+    data.create_variable(name='ypos', values=np.arange(arr.shape[1])[None, :], units=None)
+    data.transform(data.variables[0].natural_name, 'ypos')
+
+    for key, val in attrs.items():
+        data.attrs[key] = val
+
+    # finish
+    if verbose:
+        print("data created at {0}".format(data.fullpath))
+        print("  size: {0}".format(data.size))
```
Comment (ksunden) [inline WrightTools/data/_solis.py]: Is there something more informative to print? I feel like axes, shape are better than size (though size and shape will typically be the same always, at least for our current hardware, correct?)

File: WrightTools/data/_solis.py (hunk 4 above)
Comment (untzag) [reply WrightTools/data/_solis.py]: I took a look, and doesn't seem like there is a name in the footer

guess it makes sense, as I don't imagine that the software requires users to specify a name at acquisition time

File: WrightTools/data/_solis.py (hunk 5 above)
Comment (untzag) [reply WrightTools/data/_solis.py]: I agree

**Pull request merged**